### DISCUSSION TITLE
Hidden but searchable table column

### DISCUSSION THREAD

**Discussion by bilogic** [filamentphp/filament#12851] [Help]
### Package

Table builder

### Package Version

3.2.76 

### How can we help you?

Hi,

Is there a way to make a column hidden but searchable?
This is because column in question is a hasMany relationship, and thus takes up alot of width.

Any ideas? Thank you!

**Comment by dissto** [comment]
There are probably a few options you have.

You could [limit](https://filamentphp.com/docs/3.x/tables/columns/text#limiting-text-length) the text or you could make the column [wrap](https://filamentphp.com/docs/3.x/tables/columns/text#wrapping-content).

You could also pass an array of searchable columns to the `->searchable()` method.
Without knowing your case its hard to give you a tailored example but:

```php
// Assume a PostResource
TextColumn::make('title')
 ->searchable(['title', 'body'])
```

If that makes sense. It would probably be easier if you'd show some code 😁

**Reply by bilogic** [reply]
Thanks, but I want it hidden, not limit or wrap. 

I'm doing it a very hacky way now using extraAttribute CSS to stop it from displaying, but the header remains visible.

The current behavior seems to be that if a column is hidden, it is not searched, I'm hoping I missed something in the documentation.

**Reply by dissto** [reply]
Hiding a column with CSS is probably not the way to go. What prevents me from "unhiding" it 🐶

> The current behavior seems to be that if a column is hidden, it is not searched, I'm hoping I missed something in the documentation.

Yes you did miss the end of my first message 😂

**Reply by bilogic** [reply]
There is nothing special about my code :) It was copied from filament's examples, and I'm using the table search, not `isIndividual`

I have no concern about user unhiding/seeing the data, it is just the aesthetics.

**Reply by dissto** [reply]
Mhh. The code shown is not individual search either. I feel like you overlooked it as it's essentially exactly what you want 🤓

**Reply by bilogic** [reply]
Let me try it out. Thanks.

**Reply by bilogic** [reply]
```PHP
//TextColumn::make('messages.author.name'),
TextColumn::make('id')                    
    ->label('ID'),
TextColumn::make('subject')
    ->searchable(['id', 'subject'])                                       // this is ok
    ->searchable(['id', 'subject', 'messages.author.name']),              // this is not ok
```

Unfortunately it does not work, says `messages.author.name` column not found. Bear in mind `TextColumn::make('messages.author.name')` works.

**Reply by dissto** [reply]
I see In your case you probably need to modify the query and use a whereHas query. I'm on mobile right now and can't be bothered to write code. But this should nudge you in the right direction

https://github.com/filamentphp/filament/issues/5090#issuecomment-1368199186

**Reply by bilogic** [reply]
Ok thanks, let me check it out.
